neqo-transport/src/connection/mod.rs

```diff
@@ -2932,101 +2898,68 @@ impl Connection {
             _ => (),
         }
 
-        Ok(match st {
-            StreamType::UniDi => {
-                if self.indexes.remote_next_stream_uni >= self.indexes.remote_max_stream_uni {
-                    self.flow_mgr
-                        .borrow_mut()
-                        .streams_blocked(self.indexes.remote_max_stream_uni, StreamType::UniDi);
-                    qwarn!(
-                        [self],
-                        "local uni stream create blocked, next={:?} max={:?}",
-                        self.indexes.remote_next_stream_uni,
-                        self.indexes.remote_max_stream_uni
-                    );
-                    return Err(Error::StreamLimitError);
-                }
-                let new_id = self
-                    .indexes
-                    .remote_next_stream_uni
-                    .to_stream_id(StreamType::UniDi, self.role);
-                self.indexes.remote_next_stream_uni += 1;
-                let initial_max_stream_data = self
-                    .tps
-                    .borrow()
-                    .remote()
-                    .get_integer(tparams::INITIAL_MAX_STREAM_DATA_UNI);
+        match self.local_streams_fc.add_stream(st) {
+            None => {
+                self.local_streams_fc[st].blocked();
+                Err(Error::StreamLimitError)
+            }
+            Some(new_id) => Ok(match st {
+                StreamType::UniDi => {
+                    let initial_max_stream_data = self
```

Comment:
This could use some factoring.

```rust
Some(new_id) => {
let send_limit_tp = match st {
  StreamType::UniDi => tparams::INITIAL_MAX_STREAM_DATA_UNI,
  StreamType::BiDi => tparams::INITIAL_MAX_STREAM_DATA_BIDI_REMOTE,
};
let send_limit = self.tps.borrow().remote().get_integer(send_limit_tp);
self.send_streams.insert(...);
if st == StreamType::BiDi {
   // ...
}
Ok(new_id.as_u64())
}
```